File: 209-Minimum Size Subarray Sum-长度最小的子数组/209-Minimum Size Subarray Sum-长度最小的子数组.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int left = 0;
        int right = 0;
        int res = Integer.MAX_VALUE;
        int sum = 0;
        for(; right < nums.length; right++) {
        	sum += nums[right];
        	while(sum >= target) {
        		res = res < right-left + 1 ? res : right-left + 1;
        		sum -= nums[left++];
        	}
        }
        return res == Integer.MAX_VALUE ? 0 : res;
    }
}

File: 209-Minimum Size Subarray Sum-长度最小的子数组/209-Minimum Size Subarray Sum-长度最小的子数组.js
/**
 * @param {number} target
 * @param {number[]} nums
 * @return {number}
 */
var minSubArrayLen = function(target, nums) {
    let left = 0
    let right = 0
    let res = Infinity
    let sum = 0
    for(; right < nums.length; right++){
        sum += nums[right]
        while(sum >= target){
            res = Math.min(res, right - left + 1)
            sum -= nums[left++]
        }
    }
    return isFinite(res) ? res : 0
};